torchaudio/_internal/module_utils.py

```diff
@@ -0,0 +1,35 @@
+import importlib.util
+from functools import wraps
+
+
+def is_module_available(*modules: str) -> bool:
+    r"""Returns if a top-level module with :attr:`name` exists *without**
+    importing it. This is generally safer than try-catch block around a
+    `import X`. It avoids third party libraries breaking assumptions of some of
+    our tests, e.g., setting multiprocessing start method when imported
+    (see librosa/#747, torchvision/#544).
+    """
+    return all(importlib.util.find_spec(m) is not None for m in modules)
+
+
+def requires_module(*modules: str):
```

Comment:
The new decorator reads a little different
```
@_audio_backend_guard("sox")
@_mod_utils.requires_module('torchaudio._torchaudio')
```
The former makes it clear that we are requiring "sox" in particular, whereas the latter simply requires `torchaudio._torchaudio` which could give the impression this requires more, and create confusion later (even though `torchaudio._torchaudio` currently only contains "sox"). Options I see here:
* the name `torchaudio._torchaudio` is not precise enough, and should be updated.
* we move the sox specific dependency in a module properly named
* we translate a more explicit name in `requires_module` mapping "sox" to the corresponding unique module.

Thoughts?

Reply:
Yup, looks like we are on the same page.

I do agree that the previous solution was not ideal, and I am not advocating maintaining it :) It's only upside that I see was to say "sox" somewhere on the line.

That being said, I did not "request changes" to the pull request, because I also agree with you that now was likely not the time to fix this :)